<launch>
  <arg name="gui" default="true"/>

  <include file="$(find jsk_pcl_ros)/sample/sample_line_segment_detector.launch">
    <arg name="gui" value="false"/>
  </include>

  <node name="line_segment_collector"
        pkg="nodelet" type="nodelet"
        args="load jsk_pcl/LineSegmentCollector line_segment_manager">
    <remap from="~input" to="voxelgrid/output"/>
    <remap from="~input_indices" to="line_segment_detector/output/inliers"/>
    <remap from="~input_coefficients" to="line_segment_detector/output/coefficients"/>
  </node>

  <group if="$(arg gui)">
    <node name="cluster_point_indices_decomposer_detected_lines"
          pkg="nodelet" type="nodelet"
          args="load jsk_pcl/ClusterPointIndicesDecomposer line_segment_manager">
      <remap from="~input" to="voxelgrid/output"/>
      <remap from="~target" to="line_segment_detector/output/inliers"/>
    </node>

    <node name="rviz"
          pkg="rviz" type="rviz"
          args="-d $(find jsk_pcl_ros)/sample/rviz/line_segment_collector.rviz"/>
  </group>
</launch>
